homeassistant/components/hive/binary_sensor.py

```diff
@@ -10,9 +10,10 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up Hive sensor devices."""
     if discovery_info is None:
         return
-    session = hass.data.get(DATA_HIVE)
 
-    add_entities([HiveBinarySensorEntity(session, discovery_info)])
+    session = hass.data.get(DATA_HIVE)
+    for device in discovery_info:
+        add_entities([HiveBinarySensorEntity(session, device)])
```

Comment:
Please collect all entities in a list before calling `add_entities` to only call `add_entities` once.